src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala

```diff
@@ -0,0 +1,317 @@
+    sb.clear()
+
+    var sz: Int = 0
+    sz = str.length
+    var i: Int = 0
+    while (i < sz) {
+      {
+        val ch: Char = str.charAt(i)
+        if (ch > 0xfff) {
+          sb.append("\\u" + hex(ch))
+        }
+        else if (ch > 0xff) {
+          sb.append("\\u0" + hex(ch))
+        }
+        else if (ch > 0x7f) {
+          sb.append("\\u00" + hex(ch))
+        }
+        else if (ch < 32) {
+          ch match {
+            case '\b' =>
+              sb += '\\'
+              sb += 'b'
+            case '\n' =>
+              sb += '\\'
+              sb += 'n'
+            case '\t' =>
+              sb += '\\'
+              sb += 't'
+            case '\f' =>
+              sb += '\\'
+              sb += 'f'
+            case '\r' =>
+              sb += '\\'
+              sb += 'r'
+            case _ =>
+              if (ch > 0xf) {
+                sb.append("\\u00" + hex(ch))
+              }
+              else {
+                sb.append("\\u000" + hex(ch))
+              }
+          }
+        }
+        else {
+          sb.append(ch)
+        }
+      }
+      i += 1
+    }
+    sb.result()
+  }
+
+  def escapeString(str: String): String = escapeString(str, new StringBuilder(capacity = str.length * 2))
+
+  def escapeString(str: String, sb: StringBuilder): String = {
+    sb.clear()
+
+    var sz: Int = 0
+    sz = str.length
+    var i: Int = 0
+    while (i < sz) {
+      {
+        val ch: Char = str.charAt(i)
+        if (ch > 0xfff) {
+          sb.append("\\u" + hex(ch))
+        }
+        else if (ch > 0xff) {
+          sb.append("\\u0" + hex(ch))
+        }
+        else if (ch > 0x7f) {
+          sb.append("\\u00" + hex(ch))
+        }
+        else if (ch < 32) {
+          ch match {
+            case '\b' =>
+              sb += '\\'
+              sb += 'b'
+            case '\n' =>
+              sb += '\\'
+              sb += 'n'
+            case '\t' =>
+              sb += '\\'
+              sb += 't'
+            case '\f' =>
+              sb += '\\'
+              sb += 'f'
+            case '\r' =>
+              sb += '\\'
+              sb += 'r'
+            case _ =>
+              if (ch > 0xf) {
+                sb.append("\\u00" + hex(ch))
+              }
+              else {
+                sb.append("\\u000" + hex(ch))
+              }
+          }
+        }
+        else {
+          ch match {
+            case '\'' =>
+              sb += '\\'
+              sb += '\''
+            case '\"' =>
+              sb += '\\'
+              sb += '\"'
+            case '\\' =>
+              sb += '\\'
+              sb += '\\'
+            case _ =>
+              sb.append(ch)
+          }
+        }
+      }
+      i += 1
+    }
+    sb.result()
+  }
+
+  def escapeBackticked(str: String): String = escapeBackticked(str, new StringBuilder(capacity = str.length * 2))
+
+  def escapeBackticked(str: String, sb: StringBuilder): String = {
```

Comment:
As far as I can tell, this duplicates `escapeString` except with one extra case for ```\````.  I'd just add a `escapeBacktick: Boolean` flag argument.